doc/neps/nep-0057-array-in-array-out.rst

```diff
@@ -0,0 +1,231 @@
+==============================
+NEP 57 — Array In -> Array Out
+==============================
+
+:Author: Matt Haberland <mhaberla@calpoly.edu>, Add Your Name Here <example@example.com>
+:Status: Draft
+:Type: Standards Track
+:Created: 2025-05-14
+:Resolution:
+
+Abstract
+--------
+
+:ref:`NEP56` proposed adding nearly full support for the array API standard,
+but many operations involving higher rank arrays still return scalars instead
+of zero-rank arrays. This NEP would redefine the result of these operations
+to be zero-rank arrays.
```

Comment:
[nitpick] The term "rank" (in the "no. of dimensions" sense) isn't used very often in the numpy docs. Maybe because in linear algebra (and in the UK) "rank" has a different meaning? 

Either way, I suppose it would be more numpy-esque to e.g. refer to "zero-rank" as "`0-D`", "zero-dimensional", or casual "0d".

Reply:
[NEP 27](https://numpy.org/neps/nep-0027-zero-rank-arrarys.html) proposed the addition of "Zero-rank arrays" to NumPy. That seemed like a fairly appropriate source to borrow the term from.

So it's fine if the majority of authors would prefer a different term, but I don't immediately see a reason to change it.

Reply:
I would've said the same thing if I would've been around when reviewing NEP 27 🤷🏻.